tests/check_linux_layout_intel.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# Copyright (c) 2024, Intel Corporation.
+
+import argparse
+import os
+import pathlib
+
+sof_ri_info = None
+try:
+    import sof_ri_info
+except ImportError:
+    print("Note: not using sof_ri_info")
+    pass
+
+# List of Intel firmware files and their paths the upstream SOF Linux driver
+# expects to find under /lib/firmware/intel. A sof-bin release should have
+# binaries for all these targets.
+fw_list = [
```

Comment:
This should use "sets": https://docs.python.org/3/tutorial/datastructures.html#sets

Then you can print `expected_set - actual_set` and `actual_set - expected` which prints ALL differences and not just the first one found.